crates/router/src/connector/utils.rs

```diff
@@ -175,3 +176,20 @@ impl AddressDetailsData for api::AddressDetails {
             .ok_or_else(missing_field_err("address.country"))
     }
 }
+
+pub fn get_header_key_value(
+    key: &str,
+    headers: &actix_web::http::header::HeaderMap,
+) -> CustomResult<String, errors::ConnectorError> {
+    headers
```

Comment:
```suggestion
    headers
        .get(key)
        .map(|header_value| {
            header_value
                .to_str()
                .into_report()
                .change_context(errors::ConnectorError::WebhookSignatureNotFound)
        })
        .ok_or(report!(errors::ConnectorError::WebhookSourceVerificationFailed))?
```

Reply:
`?` operator cannot convert from `Result<&str, _>` to `Result<std::string::String, _>`
`map(String::from)` is required

Reply:
the return type need not be string it can be slice

Reply:
yes